### DISCUSSION TITLE
Users of one organization by default shouldn't be allowed to log in other organizations

### DISCUSSION THREAD

**Discussion by hanspoo** [zitadel/zitadel#8329] [General]
Hi, we are moving to zitadel from keycloak, i like zitadel user experience and features are great, thanks, but we are stuck, really stuck, when testing security in a multi organization project, we've found that users created in one organization are able to log in other organizations, this breaks all our asumptions, we've not found how to configure or stop this, it should be something easy and please give a help.

Thanks for reading.

**Comment by hifabienne** [comment]
Hei @hanspoo 
Yes you are right, but there are some possibilities. 
Good to know is that we destinct between authentication and authorization, the login is basically only the authentication part, without checking for any permissions, as we recommend putting the permission check into your application.
Per default we show always the login with the settings from the default organization, but all users are able to login. As soon as we have identitfied the user, we can also render the settings from their own organization.

1. You can send the organization id scope to the login, this will trigger the login from an organization directly and also trigger the settings from that organization urn:zitadel:iam:org:roles:id:{orgID} https://zitadel.com/docs/apis/openidoauth/scopes#reserved-scopes
2. You can also restrict the access through permissions. Example you have a project A with the app A in organization A. You can enable the "Check authorization on Authentication" or "Check for Project on Authentication" on the project, if you redirect from app a to the login now only users with either an authorization for the project or where the organization itself has the project are allowed to login. This also allows you to have delegated accessmanagement. if you grant the project to another organization, the users of that org would also be allowed to login
<img width="759" alt="image" src="https://github.com/user-attachments/assets/9dd589f6-86e4-49f0-9637-04ac81be9eb6">

Let me know if that would solve your problem

**Reply by hanspoo** [reply]
Hi, thanks for the fast reply !! i have a practical issue: i'm running the latest docker image, and can't find the options you show in the screen shoot,  can you give me a clue ?

**Reply by hifabienne** [reply]
the management console should be deployed with ZITADEL you can access it with your domain/ui/console
otherwise you can also manipulate it through the API: https://zitadel.com/docs/apis/resources/mgmt/management-service-update-project

**Comment by hanspoo** [comment]
HI, thanks for your time. I was looking in the Organization settings, and it was in project settings, we are testing now, thank you.
